sdcm/nemesis.py

```diff
@@ -1710,6 +1710,93 @@ def disrupt_network_start_stop_interface(self):  # pylint: disable=invalid-name
         finally:
             self.target_node.remoter.run("sudo /sbin/ifup eth1")
 
+    def streaming_err_and_rebuild(self, task='decommission'):
+        """
+        Stop streaming task in middle and rebuild the data on the node.
+        """
+
+        def streaming_task_thread(nodetool_task='decommission'):
+            """
+            Execute some nodetool command to start persistent streaming
+            task: decommission | rebuild | repair
+            """
+            try:
+                self.target_node.run_nodetool(nodetool_task)
+            except Exception:  # pylint: disable=broad-except
+                self.log.debug('%s is stopped' % nodetool_task)
+
+        streaming_thread = threading.Thread(target=streaming_task_thread, kwargs={'nodetool_task': task})
+        streaming_thread.start()
+
+        def is_streaming_started(nodetool_task='decommission'):
+            start_logs = {'rebuild': 'Rebuild starts',
+                          'decommission': 'DECOMMISSIONING: starts', 'repair': 'Repair start'}
+            result = self.target_node.remoter.run(
+                'ps aux|grep nodetool.*%s |grep -v grep' % nodetool_task, ignore_status=True)
+            self.log.debug(result.stdout)
+            if nodetool_task not in result.stdout:
+                return False
+            streaming_logs = self.target_node.search_database_log(start_logs[nodetool_task], start_from_beginning=False)
+            self.log.debug(streaming_logs)
+            return len(streaming_logs) > 0
+
+        wait.wait_for(func=is_streaming_started, timeout=200, step=1,
+                      text='Wait for streaming starts', throw_exc=False, nodetool_task=task)
+
+        self.log.info('Interrupt the task (%s) to trigger streaming error' % task)
+        if random.randint(0, 1) == 0:
+            self.log.debug('Interrupt the streaming task by pkill')
+            self.target_node.remoter.run('pkill -9 -f %s' % task)
+        else:
+            self.log.debug('Interrupt the streaming task by hard reboot')
+            self.target_node.reboot(hard=True, verify_ssh=True)
+        streaming_thread.join(60)
+
+        ips = []
+        seed_nodes = [node for node in self.cluster.nodes if node.is_seed]
+        status = self.cluster.get_nodetool_status(verification_node=seed_nodes[0])
+        self.log.debug(status)
+        for dc_info in status.values():
+            try:
+                ips.extend(dc_info.keys())
+            except Exception:  # pylint: disable=broad-except
+                self.log.debug(dc_info)
+
+        if self.target_node.ip_address not in ips:
+            self.log.error(
+                'The target node is decommission unexpectedly, decommission might complete before stopping it. Re-add a new node')
```

Comment:
Assuming what happening is not good enough, try to search the log for messages and then do the interruptions.
Or support new events and look for these events.

Reply:
changed to check the log 'DECOMMISSIONING: done'